src/main/kotlin/traffic_simulation/RoadNetwork.kt

```diff
@@ -1,43 +1,56 @@
 package traffic_simulation
 
+import java.util.*
+
 class RoadNetwork(val capacity: Int) {
 
-    fun calculateDemandForTimestepAndCauseTrafficJam(timestep: Int, vehicleList: List<Vehicle>): Int{
-        var demandAtTimestep: Int =0
+    fun calculateDemandForTimestepAndCauseTrafficJam(timestep: Int, vehicleList: List<Vehicle>): Int {
+        var demandAtTimestep: Int = 0
 
-        for(vehicle in vehicleList){
-            if(vehicle.vehicleWantsToDriveAt(timestep)){
-                demandAtTimestep = demandAtTimestep +1
+        for (vehicle in vehicleList) {
+            if (vehicle.vehicleWantsToDriveAt(timestep)) {
+                demandAtTimestep = demandAtTimestep + 1
             }
         }
 
         val probabilityOfDelay: Int
         val percentOfCapacityUsage: Int = (demandAtTimestep / capacity) * 100
-        when(percentOfCapacityUsage){
+        when (percentOfCapacityUsage) {
+            !in Int.MIN_VALUE..Int.MAX_VALUE -> println("insane road")
             in 0..49 -> probabilityOfDelay = 5
-            in 51..69 -> probabilityOfDelay = 10
+            in 50..69 -> probabilityOfDelay = 10
             in 70..89 -> probabilityOfDelay = 20
             in 90..109 -> probabilityOfDelay = 30
             in 110..124 -> probabilityOfDelay = 50
             in 125..149 -> probabilityOfDelay = 70
             in 150..Int.MAX_VALUE -> probabilityOfDelay = 90
         }
-        // The Random-parts ensue.
 
-        if(demandAtTimestep>capacity){
-            for(vehicle in vehicleList){
+        fun DelayedByProbability(probability: Int): Boolean {
```

Comment:
Die Funktionsdeklaration hast du gerade noch in einer anderen Funktion drinnen. Fände es besser wenn die Funktion außerhalb wäre. Aber mit der kurzen Version von @nicolaiparlog brächte man die Funktion ja garnicht mehr.